src/js/components/KeptEntry.jsx

```diff
@@ -61,34 +60,34 @@ var KeptEntry = React.createClass({
     event.preventDefault();
     this.unhighlight();
     var newIndex = parseInt(event.dataTransfer.getData("text/plain"), 10);
-    this.props.move(newIndex, this.props.key);
+    this.props.move(newIndex, this.props.itemIndex);
   },
 
   highlight: function() {
-    this.getDOMNode().querySelector(".panel").classList.add("targetted");
+    reactDom.findDOMNode(this.refs.panel).classList.add("targetted");
   },
 
   unhighlight: function() {
-    this.getDOMNode().querySelector(".panel").classList.remove("targetted");
+    reactDom.findDOMNode(this.refs.panel).classList.remove("targetted");
```

Comment:
I think I've read in the [changelog for 0.14](http://facebook.github.io/react/blog/2015/09/10/react-v0.14-rc1.html) that you can now refer to the DOM node directly from the ref:

> The other big change we’re making in this release is exposing refs to DOM components as the DOM node itself. That means: we looked at what you can do with a ref to a React DOM component and realized that the only useful thing you can do with it is call `this.refs.giraffe.getDOMNode()` to get the underlying DOM node. In this release, `this.refs.giraffe` is the actual DOM node.